dataprofiler/profilers/order_column_profile.py

```diff
@@ -1,15 +1,29 @@
 """Index profile analysis for individual col within structured profiling."""
 from __future__ import annotations
 
-from typing import cast
+from abc import abstractmethod
+from typing import Protocol, TypeVar
 
+import numpy as np
 from pandas import DataFrame, Series
 
 from . import utils
 from .base_column_profilers import BaseColumnProfiler
 from .profiler_options import OrderOptions
 
 
+class Comparable(Protocol):
+    """Protocol for ensuring comparable types, in this case both floats or strings."""
+
+    @abstractmethod
+    def __lt__(self: CT, other: CT) -> bool:
+        """Protocol for ensuring comparable values."""
+        pass
+
+
+CT = TypeVar("CT", bound=Comparable)
+
+
```

Comment:
Allows for comparable types. Specifying float | str would error as you can't do a float + str.